content/v2.0/actions/http-caching.md

```diff
@@ -0,0 +1,165 @@
+---
+title: HTTP Caching
+order: 120
+---
+
+Actions provide several features to help you make use of [HTTP caching][mdn-http-caching].
+
+[mdn-http-caching]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching
+
+## Cache control
+
+The [Cache-Control response header][mdn-cache-control] is the main way to control HTTP caching behavior. You can use the `#cache_control` method on your action's response object to set this header.
+
+```ruby
+# app/actions/books/index.rb
+
+module Bookshelf
+  module Actions
+    module Books
+      class Index < Bookshelf::Action
+        def handle(request, response)
+          # Sets `Cache-Control: public, max-age: 600`
+          response.cache_control :public, max_age: 600
+        end
+      end
+    end
+  end
+end
+```
+
+The `cache_control` method accepts one or more of the following [Cache-Control directives][mdn-cache-control-directives] as its arguments.
+
+- `:public`
+- `:private`
+- `:no_cache`
+- `:no_store`
+- `:must_validate`
+- `:proxy_revalidate`
+- `max_age: N` (`N` here and below should be an integer representing a period in seconds)
+- `s_maxage: N`
+- `min_fresh: N`
+- `max_stale: N`
+
+[mdn-cache-control]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
+[mdn-cache-control-directives]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#cache_directives
+
+## Expires
+
+The [Expires response header][mdn-expires] sets the date and time after which the response is considered expired. The Expires header is an older standard, and Cache-Control is preferred by modern browsers.
+
+You can use the `#expires` method on your action's response object to set this header along with a matching Cache-Control header.
+
+```ruby
+# app/actions/books/index.rb
+
+module Bookshelf
+  module Actions
+    module Books
+      class Index < Bookshelf::Action
+        def handle(request, response)
+          # Sets `Expires: Sun, 20 Nov 2022 17:47:02 GMT, Cache-Control: public, max-age=600`
+          response.expires 60, :public, max_age: 600
+        end
+      end
+    end
+  end
+end
+```
+
+## Conditional requests
+
+[HTTP conditional requests][mdn-conditional-requests] are an interaction between the server and a client wherein the result of a request can be changed by comparing the related server resource with the value of a validator. These requests can be useful to validate the content of a client's cache for the given resource.
+
+For example, for a conditional `GET` request, there is a two-step workflow:
+
+1. The response to an initial request includes a “validator” in its headers, either the date of last modification for the resource (a Last-Modified response header and a subsequent If-Modified-Since request header), or an string identifying the version of the resource (an ETag  response header and a subsequent If-None-Match request header)
```

Comment:
```suggestion
1. The response to an initial request includes a “validator” in its headers, either the date of last modification for the resource (a Last-Modified response header and a subsequent If-Modified-Since request header), or an string identifying the version of the resource (an ETag response header and a subsequent If-None-Match request header)
```